engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/new/new-repository.controller.js

```diff
@@ -17,144 +17,156 @@
-              /* eslint-disable camelcase */
-                $scope.repository.ansible_collection_requirements = data;
-                $scope.$apply();
-            });
-            reader.readAsText(element.files[0]);
-        };
-
-        ContentCredential.queryUnpaged(function (contentCredentials) {
-            $scope.contentCredentials = contentCredentials.results;
-        });
-
-        Architecture.queryUnpaged(function (architecture) {
-            var results = architecture.results;
-            var noarch = {
-                id: 'noarch',
-                name: translate('Default'),
-                value: null
+            $scope.handleFiles = function (element) {
+                var reader = new FileReader();
+                reader.addEventListener("loadend", function() {
+                    var data = reader.result;
+                    /* eslint-disable camelcase */
+                    $scope.repository.ansible_collection_requirements = data;
+                    $scope.$apply();
+                });
+                reader.readAsText(element.files[0]);
             };
-            results.map(function(i) {
-                i.id = i.name;
+
+            ContentCredential.queryUnpaged(function (contentCredentials) {
+                $scope.contentCredentials = contentCredentials.results;
             });
-            results.unshift(noarch);
-            $scope.architecture = results;
-            $scope.repository.arch = results[0].id;
-        });
-
-        $scope.save = function (repository) {
-            if (repository.content_type === 'ostree') {
-                repository.unprotected = false;
-            }
-            if (repository.content_type !== 'yum') {
-                repository['download_policy'] = '';
-            }
-            if (repository.arch === 'Default') {
-                repository.arch = null;
-            }
-            repository.$save(success, error);
-        };
 
-        $scope.repository['http_proxy_policy'] = HttpProxyPolicy.policies[0].label;
-        $scope.policies = HttpProxyPolicy.policies;
-        $scope.proxies = [];
+            Architecture.queryUnpaged(function (architecture) {
+                var results = architecture.results;
+                var noarch = {
+                    id: 'noarch',
+                    name: translate('Default'),
+                    value: null
+                };
+                results.map(function(i) {
+                    i.id = i.name;
+                });
+                results.unshift(noarch);
+                $scope.architecture = results;
+                $scope.repository.arch = results[0].id;
+            });
+
+            $scope.save = function (repository) {
+                if (repository.content_type === 'ostree') {
+                    repository.unprotected = false;
+                }
+                if (repository.content_type === 'yum') {
+                    repository.required_tags = $scope.requiredTagsParam();
```

Comment:
same here, I don't know why the diff is so weird.  Only real changes to this file are here and the requiredTags methods

Reply:
thank goodness for 'Hide whitespace changes'